073-improve-handling-of-CA-renewals-and-replacements-in-client-based-operands.md

```diff
@@ -0,0 +1,211 @@
+data:
+  ca1.crt: <CA-1>
+  ca2.crt: <CA-2>
+  ca3.crt: <CA-3>
+```
+
+The user can load all three CAs with the following configuration:
+```yaml
+  tls:
+    trustedCertificates:
+      - secretName: my-secret
+        extension: crt
+```
+
+As today, when one of the existing certificates changes, Strimzi will automatically toll the operand to update the trusted certificates.
+The same will happen also when a new record is added to the Secret that matches the extension - e.g. `ca-4.crt`.
+
+In the scenario described in the _Motivation_ section, the new APi will work like this:
+1. Strimzi generates the new CA public and private keys and stores the new CA public key as `ca.crt` in the Kubernetes Secret.
+   The public key of the old CA is renamed to `ca-<timestamp>.crt`.
+2. Strimzi starts rolling the Kafka brokers to make sure they trust the new CA.
+   At this point in time, the Kafka brokers still use the old server certificates signed by the old CA.
+3. The Kafka client based operand (e.g. Kafka Connect cluster) using the new `.spec.tls` configuration:
+   ```yaml
+   tls:
+      trustedCertificates:
+         - secretName: my-cluster-cluster-ca-cert
+           extesion: crt
+   ```
+   Will see that the `ca.crt` in the Secret was updated and that a new `ca-<timestamp>.crt` was added.
+   That will trigger a rolling update of the operand to use the new CA.
+   After the rolling update, it will trust both the old and new CAs and continue working without any problems while the CA is being replaced.
+4. After all brokers are rolled to trust the new CA, another rolling update happens and a new server certificates signed by the new CA are rolled out to the Kafka brokers in another rolling update.
+5. Once all brokers are rolled once again and use the new CA, the operand such as Kafka Connect will be able to connect again and start working.
+   The impact and length of the outage will depend on many different aspects:
+   * How many nodes does the operand such as Kafka Connect have.
+     If it is only one replica, the whole operand will be down.
+     With multiple replicas, only one replica should be down.
+   * How big and busy the Kafka cluster is.
+     A small cluster might take few minutes to move to the new CA completely.
+     A big cluster with many nodes that are busy and need more time to re-sync between restarts might take much longer.
+6. After all brokers (and other components such as ZooKeeper, Topic and User Operator, Cruise Control etc.) use the new CA, another rolling update will be done to completely remove the trust to the old CA by removing the `ca-<timestamp>.crt` from the Secret and rolling all components once again.
+   This last rolling update has currently no impact on the operands such as Kafka Connect.
+7. The removal of `ca-<timestamp>.crt` from the Secret will cause rolling update of the client-based operand (e.g. Connect).
+   This rolling update will remove the trust to the old CA from the operand as well.
```

Comment:
I wonder if it would be useful to summarise the difference:  _In the updated scenario, the solution allows the operand to trust the old and new CA during the renewal process and removes the trust to the old CA from the operand on completion._

Reply:
Yeah, I think this makes sense. I tried to make it more clear that this rolling update is new when using the new API.